pkg/discovery-manager/discovery_controller.go

```diff
@@ -47,13 +56,276 @@ type DiscoveryReconciler struct {
+		discovery.Status.Phase.Phase != nodecorev1alpha1.PhaseRunning &&
+		discovery.Status.Phase.Phase != nodecorev1alpha1.PhaseIdle {
+
+		setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseRunning)
+
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Discovery %s status before reconcile: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	switch discovery.Status.Phase.Phase {
+	case nodecorev1alpha1.PhaseRunning:
+		flavoursCR, err := getFlavour(discovery.Spec.Selector)
+		if err != nil {
+			klog.Errorf("Error when getting Flavour: %s", err)
+			return ctrl.Result{}, err
+		}
+
+		klog.Infof("Flavours found: %d", len(flavoursCR))
+
+		// first is a boolean that is used to check if the current flavour is the first one in the array
+		var first bool = true
+		var i int = 1
+		for _, flavourCR := range flavoursCR {
+			if first {
+				// We refer to the first peering candidate as the one that is reserved
+				peeringCandidate = createPeeringCandidateCustomResources(flavourCR, &discovery, first, i)
+				klog.Infof("Peering Candidate %s created", peeringCandidate.Name)
+			} else {
+				// the others are created as not reserved
+				peeringCandidateNotReserved := createPeeringCandidateCustomResources(flavourCR, &discovery, first, i)
+				klog.Infof("Peering Candidate %s created but not reserved", peeringCandidateNotReserved.Name)
+				err = r.Create(context.Background(), peeringCandidateNotReserved)
+				if err != nil {
+					klog.Errorf("Error when creating Peering Candidate not reserved: %s", err)
+				}
+			}
+			i++
+			first = false
+		}
+
+		err = r.Create(context.Background(), peeringCandidate)
+		if err != nil {
+			klog.Errorf("Error when creating Peering Candidate: %s", err)
+			setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseFailed)
+		} else {
+			// Update the Discovery with the PeeringCandidate
+			discovery.Spec.PeeringCandidate.Name = peeringCandidate.Name
+			discovery.Spec.PeeringCandidate.Namespace = DEFAULT_NAMESPACE
+			if err := r.Update(ctx, &discovery); err != nil {
+				klog.Errorf("Error when updating Discovery %s: %s", discovery.Name, err)
+				return ctrl.Result{}, err
+			}
+			klog.Infof("Discovery %s updated", discovery.Name)
+			setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseSolved)
+		}
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Solver %s status: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		} else {
+			// Create a Reservation CR
+			reservationCR := createReservationCustomResource(peeringCandidate)
+			err := r.Create(context.Background(), reservationCR)
+			if err != nil {
+				klog.Errorf("Error when creating Reservation: %s", err)
+				setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseFailed)
+			} else {
+				klog.Infof("Reservation %s created", reservationCR.Name)
+			}
```

Comment:
The DiscoveryManager is not the responsible for the creation of the reservation! do you remember?

Reply:
I know, I just create a CR so that the Contract Manager can trigger a reserve.. otherwise how can I do it?

Reply:
AFTER CALL: this was a work for Alessandro, I comment all these functions with ///TODO: for Ale, move this function etc..

Reply:
New logic implemented.